packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -254,8 +297,28 @@ export class BucketDeployment extends CoreConstruct {
       uuid += `-${memoryLimit.toString()}MiB`;
     }
 
+    // if user specify to use VPC, define another singleton handler
+    // with this configuration. otherwise, it won't be possible to use multiple
+    // configurations since we have a singleton.
+    // A VPC is a must if EFS storage is used and that's why we are only using VPC in uuid.
+    if (vpc) {
+      uuid += `-${vpc.node.addr.toUpperCase()}`;
+    }
+
     return uuid;
   }
+
+  /**
+   * Function to get/create a stack singleton instance of EFS FileSystem per vpc.
+   *
+   * @param scope Construct
+   * @param fileSystemProps EFS FileSystemProps
+   */
+  private getOrCreateEfsFileSystem(scope: Construct, fileSystemProps: efs.FileSystemProps): efs.FileSystem {
+    const stack = cdk.Stack.of(scope);
+    const uuid = `Efs-${fileSystemProps.vpc.node.addr.toUpperCase()}`;
```

Comment:
I think it makes sense to generate this uid using the same `renderSingletonUuid` function used to generate the lambda singleton UID - this way we guarantee a 1:1 relationship between a lambda function and a file system.

Reply:
This method is returning the node in addition to `renderSingleteUuid`.  Having `memoryLimit` in argument will create new instance of EFS per Lambda function which looks undesirable. With above implementation, all Lambda instances in same vpc will use singlge EFS volume.

Do we want to have different EFS volume for different instances of lambda in same VPC?

Reply:
I see. Ok I agree there's no point in creating multiple EFS 👍 

Lets give it a more descriptive name though:

```suggestion
    const uuid = `BucketDeploymentEFS-VPC${fileSystemProps.vpc.node.addr.toUpperCase()}`;
```

Reply:
Here as well, I would not alter the case of `node.addr`

```suggestion
    const uuid = `Efs-${fileSystemProps.vpc.node.addr}`;
```

Reply:
Updated as suggested.